### DISCUSSION TITLE
can i escape from infinite loop?

### DISCUSSION THREAD

**Discussion by tony-jinwoo-ahn** [pyodide/pyodide#3891] [Q&A] [answered]
if i run runPythonAsync(code); in js code and if the code contains infinite loop like this:
```
while True:
  time.sleep(1)
```
if another user runs this code snippet in my website, then browser freezes.
i know that users should run code under there own risks, but i'd like to provide `timeout` to help them escape from infinite loop.
how can I add timeout when runPythonAsync()?

**Comment by ryanking13** [comment, marked as answer]
You can interrupt the execution, but only in the WebWorker (https://pyodide.org/en/stable/usage/keyboard-interrupts.html)

**Reply by tony-jinwoo-ahn** [reply]
Thank you very much :)
